src/hooks.js

```diff
@@ -0,0 +1,60 @@
+import { useEffect, useRef, useState } from 'react'
+import { getDog } from './utils/fetch-request'
+
+export const useFirstMount = () => {
+  const isFirstMount = useRef(true)
+
+  useEffect(() => {
+    return () => {
+      isFirstMount.current = false
+    }
+  }, [])
+
+  return isFirstMount
+}
+
+export const useInitDog = () => {
+  const [initDogs, setInitDogs] = useState([])
+  const isFirstMount = useFirstMount()
+
+  useEffect(() => {
+    if (isFirstMount.current) {
+      Promise.all([getDog(), getDog(), getDog()])
+        .then(dogs => {
+          setInitDogs(() => {
+            return dogs.map(dog => dog.message)
+          })
+        })
+    }
+  }, [isFirstMount])
+
+  return initDogs
+}
+
+
+export const usePreloadDog = () => {
+  const [preloadDogs, setPreloadDogs] = useState([])
+  const isFirstMount = useFirstMount()
+
+  useEffect(() => {
+    if (!isFirstMount.current) {
```

Comment:
Is this the issue...?

Looks like `console.info('Run one time')` is in the wrong place...?

`if (!isFirstMount.current) {` means if it is NOT the first mount - or in other words - every renders after the first one

Reply:
@wood-crafter

Reply:
> Is this the issue...?
> 
> Looks like `console.info('Run one time')` is in the wrong place...?
> 
> `if (!isFirstMount.current) {` means if it is NOT the first mount - or in other words - every renders after the first one

@hoangph271 
- im using console to count time of running if (!isFirstMount.current) on completely rendered
- `if (!isFirstMount.current) { means if it is NOT the first mount - or in other words - every renders after the first one` - Yes